src/schema/validator.js

```diff
@@ -4,28 +4,185 @@
  * Copying and unauthorised use of this material strictly prohibited.
  */
 
-const Ajv = require('ajv');
-const EPCISDocument = require('./EPCISDocument.schema.json');
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import * as fieldNames from '../utils/field-names';
+import definitions from './definitions.json';
+import validationMode from '../settings';
+import EPCISDocument from './EPCISDocument.schema.json';
+import ObjectEvent from './ObjectEvent.schema.json';
+import AggregationEvent from './AggregationEvent.schema.json';
+import TransformationEvent from './TransformationEvent.schema.json';
+import TransactionEvent from './TransactionEvent.schema.json';
+import AssociationEvent from './AssociationEvent.schema.json';
+import ExtendedEvent from './ExtendedEvent.schema.json';
 
-const ajv = new Ajv({ useDefaults: true });
+/**
+ * @typedef {object} ValidatorResult
+ * @property {boolean} success - true if the validator found no errors.
+ * @property {Array<string>} errors - List of errors.
+ */
+const successResult = { success: true, errors: [] };
+
+const ajv = addFormats(new Ajv({ useDefaults: true, strict: false }), { mode: validationMode });
+
+/**
+ * Load a schema and include 'definitions'.
+ *
+ * @param {object} schema - schema to be loaded.
+ * @returns {object} Loaded and enhanced schema.
+ */
+export const loadSchema = ((schema) => ({ ...schema, definitions })
+);
 
 /** Available schemas */
 const validators = {
   EPCISDocument: ajv.compile(EPCISDocument),
+  ObjectEvent: ajv.compile(loadSchema(ObjectEvent)),
+  AggregationEvent: ajv.compile(loadSchema(AggregationEvent)),
+  TransformationEvent: ajv.compile(loadSchema(TransformationEvent)),
+  TransactionEvent: ajv.compile(loadSchema(TransactionEvent)),
+  AssociationEvent: ajv.compile(loadSchema(AssociationEvent)),
+  ExtendedEvent: ajv.compile(loadSchema(ExtendedEvent)),
 };
 
 /**
- * Validate an EPCIS document.
+ * Validate data against a nominated schema.
  *
- * @param {object} instance - The data to validate against the schema.
- * @returns {Array<string>} Any errors.
+ * @param {object} data - Data to validate.
+ * @param {string} schemaName - Name of the schema from validatorNames.
```

Comment:
I would create another function called validateEvent. The second parameter will be  called 'eventType' - e.g 'ObjectEvent'.

Because it isn't easy for someone that doesn't know the SDK to know what to put in `schemaName`